Commit ID: 1fc7996c3977724c75901d385f910a3bcb521a61
Change ID: rromutrvsuvnklkwtompyoopszqkontp
Author   : flukey <flukey@vapourmail.eu> (2026-10-05 15:32:40)
Committer: flukey <flukey@vapourmail.eu> (2026-10-05 15:58:35)

    fix stdio usage

diff --git a/lib/commands/toolcontext.c b/lib/commands/toolcontext.c
index f96441c91a..44fa0ccc0d 100644
--- a/lib/commands/toolcontext.c
+++ b/lib/commands/toolcontext.c
@@ -1684,6 +1684,7 @@
 	reset_lvm_errno(1);
 
 	/* Set in/out stream buffering before glibc */
+#ifdef __GLIBC__
 	if (set_buffering
 	    && !cmd->running_on_valgrind /* Skipping within valgrind execution. */
 #ifdef SYS_gettid
@@ -1727,6 +1728,7 @@
 	} else if (!set_buffering)
 		/* Without buffering, must not use stdin/stdout */
 		init_silent(1);
+#endif
 
 	/*
 	 * Environment variable LVM_SYSTEM_DIR overrides this below.
@@ -2065,6 +2067,7 @@
 	if (cmd->cft_def_hash)
 		dm_hash_destroy(cmd->cft_def_hash);
 
+#ifdef __GLIBC__
 	if (!cmd->running_on_valgrind && cmd->linebuffer) {
 		int flags;
 		/* Reset stream buffering to defaults */
@@ -2088,6 +2091,7 @@
 
 		free(cmd->linebuffer);
 	}
+#endif
 
 	destroy_config_context(cmd);
 
diff --git a/tools/lvmcmdline.c b/tools/lvmcmdline.c
index 0499ffe77d..fe5942e59e 100644
--- a/tools/lvmcmdline.c
+++ b/tools/lvmcmdline.c
@@ -3400,7 +3400,7 @@
 	int err = is_valid_fd(STDERR_FILENO);
 
 	if (!is_valid_fd(STDIN_FILENO) &&
-	    !(stdin = fopen(_PATH_DEVNULL, "r"))) {
+	    !freopen(_PATH_DEVNULL, "r", stdin)) {
 		if (err)
 			perror("stdin stream open");
 		else
@@ -3410,7 +3410,7 @@
 	}
 
 	if (!is_valid_fd(STDOUT_FILENO) &&
-	    !(stdout = fopen(_PATH_DEVNULL, "w"))) {
+	    !freopen(_PATH_DEVNULL, "w", stdout)) {
 		if (err)
 			perror("stdout stream open");
 		/* else no stdout */
@@ -3418,7 +3418,7 @@
 	}
 
 	if (!is_valid_fd(STDERR_FILENO) &&
-	    !(stderr = fopen(_PATH_DEVNULL, "w"))) {
+	    !freopen(_PATH_DEVNULL, "w", stderr)) {
 		printf("stderr stream open: %s\n",
 		       strerror(errno));
 		return 0;
